### PR TITLE
[circleci] Conditional tests

### PR DESCRIPTION
**Pull request by stephenroller**
An attempt to make tests a bit smarter, where gpu, long cpu, and mturk tests are all run conditionally from CircleCI.

### CODE REVIEW

**Comment by stephenroller** [comment]
@KyleTryon tried the v2 in f6ff9c9. It looks like that worked pretty well!

**Comment by stephenroller** [comment]
Ready for review.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,10 +10,17 @@ We actively welcome your pull requests.
...
-5. Make sure your code lints (`bash tests/lint_changed.sh`)
-6. If you haven't already, complete the Contributor License Agreement ("CLA").
+4. Make sure your code lints (`bash tests/lint_changed.sh`)
+5. Ensure the test suite passes. Run `python setup.py test`.
+6. If you've added a new dataset, you should also run
+   `python setup.py test -s tests.suites.datatests`. Copy-paste the output into a
+   comment in your PR.
+7. If you haven't already, complete the Contributor License Agreement ("CLA").
+
+Our continuous integration suite will run only fast tests before merging. We also
```
Comment (klshuster) [inline CONTRIBUTING.md]: Thanks this is exactly what i was referring to

Comment (stephenroller) [reply CONTRIBUTING.md]: ya i went and updated it after you pointed it out :P

**Pull request merged**